### PR TITLE
Add a userCommand NetPerfToJson

### PR DESCRIPTION
**Pull request by vancem**
This converts a .NetPerf file to a *.netperf.json file that should be very easy to render in tools like speedscope.

You can convert a file with the command
```
    perfView userCommand NetPerfToJson  FILE.netperf
```
It will generate a FILE.netperf.json file.

This format is still a bit experimental.  It will probably be tweeked.

### CODE REVIEW

**Comment by mjsabby** [comment]
@vancem you may want to remerge with master because its' too difficult to review with all the changes.

**Comment by vancem** [comment]
@mjsabby  fixed merge issue

File: src/PerfView/OtherSources/XMLStackSource.cs [hunk 1]
```diff
@@ -113,6 +113,85 @@ public static void WriteStacks(StackSource source, XmlWriter writer)
...
+        }
+        public static void WriteStackViewAsJson(StackSource source, string fileName, Action<TextWriter> additionalData = null)
+        {
+            if (File.Exists(fileName))
+                File.Delete(fileName);
+
+            using (var writeStream = File.CreateText(fileName))
+                WriteStackViewAsJson(source, writeStream, additionalData);
+        }
+        public static void WriteStackViewAsJson(StackSource source, TextWriter writer, Action<TextWriter> additionalData = null)
```
Comment (mjsabby) [inline src/PerfView/OtherSources/XMLStackSource.cs]: newline missing between the methods.

File: src/PerfView/OtherSources/XMLStackSource.cs [hunk 2]
```diff
@@ -113,6 +113,85 @@ public static void WriteStacks(StackSource source, XmlWriter writer)
         }
     }
 
+    public class JsonStackSourceWriter
+    {
+        public static void WriteStackViewAsZippedJson(StackSource source, string fileName, Action<TextWriter> additionalData = null)
+        {
+            if (File.Exists(fileName))
+                File.Delete(fileName);
```
Comment (mjsabby) [inline src/PerfView/OtherSources/XMLStackSource.cs]: curly braces missing, not sure if there's a standard in this repo yet, but it'd be great to have a build failure for this one.

**Pull request closed without merging**